From: "Daniel P. Berrange" <berrange@redhat.com>
Date: Tue, 11 Oct 2011 11:26:59 +0100
Subject: Fix storage pool source comparison to avoid comparing with self

If we are comparing storage pools we must skip comparing with
ourself, so that re-defining an existing pool works

* conf/storage_conf.c: Skip self when comparing

---
 src/conf/storage_conf.c |    4 ++++
 1 files changed, 4 insertions(+), 0 deletions(-)

diff --git a/src/conf/storage_conf.c b/src/conf/storage_conf.c
index e893b2d..eb39198 100644
--- a/src/conf/storage_conf.c
+++ b/src/conf/storage_conf.c
@@ -1730,6 +1730,10 @@ int virStoragePoolSourceFindDuplicate(virStoragePoolObjListPtr pools,
         if (def->type != pool->def->type)
             continue;
 
+        /* Don't mach against ourself if re-defining existing pool ! */
+        if (STREQ(pool->def->name, def->name))
+            continue;
+
         virStoragePoolObjLock(pool);
 
         switch (pool->def->type) {
-- 
